tests/pass/float.rs

```diff
@@ -29,15 +32,71 @@ fn main() {
     test_algebraic();
 }
 
-// Helper function to avoid promotion so that this tests "run-time" casts, not CTFE.
-// Doesn't make a big difference when running this in Miri, but it means we can compare this
-// with the LLVM backend by running `rustc -Zmir-opt-level=0 -Zsaturating-float-casts`.
-#[track_caller]
-#[inline(never)]
-fn assert_eq<T: PartialEq + Debug>(x: T, y: T) {
-    assert_eq!(x, y);
+trait Float: Copy + PartialEq + Debug {
+    /// The unsigned integer with the same bit width as this float
+    type Int: Int;
+    const BITS: u32;
+    const EXPONENT_BITS: u32;
+    const SIGNIFICAND_BITS: u32 = Self::BITS - Self::EXPONENT_BITS - 1;
```

Comment:
libcore has MANTISSA_DIGITS; this can be computed from them, can't it?

Reply:
Yes, updated to use that